ui/app/models/config-ui/message.js

```diff
@@ -11,10 +11,21 @@ import { withFormFields } from 'vault/decorators/model-form-fields';
 const validations = {
   title: [{ type: 'presence', message: 'Title is required.' }],
   message: [{ type: 'presence', message: 'Message is required.' }],
+  link: [
+    {
+      validator(model) {
+        const linkTitle = Object.keys(model.link);
+        const [linkHref] = Object.values(model.link);
+        if (!model?.link || linkTitle.length < 1 || linkTitle[0] === '') return true;
```

Comment:
If link is undefined then I think `Object.keys(model.link);` will fail because that's `Object.keys(undefined);` 

So I think `!model?.link` still needs to be the first conditional, then the second part could be

```js
 validator(model) {
        if ((!model?.link) return true
        const [linkTitle] = Object.keys(model.link); // linkTitle could be undefined or an empty string, both are falsy
        const [linkHref] = Object.values(model.link);
        if (!linkTitle) return true 
        return !!linkHref;
},

Reply:
Here's that demo'd in the console to make sure I'm thinking of the logic correctly
<img width="428" alt="Screenshot 2024-01-18 at 2 25 01 PM" src="https://github.com/hashicorp/vault/assets/68122737/4dd35b54-c35c-47b2-8aeb-512ef342018f">